documentation/projects/proposals/infrastructure_improvements/20240105_implementation_plan_move_remaining_services_to_openverse_org.md

```diff
@@ -0,0 +1,738 @@
+
+There are also two supplemental parts, a [preliminary step](#0-preliminary-work)
+and a
+[finalisation step](#3-update-references-announce-to-make-blog-and-finalise).
+
+## Expected Outcomes
+
+<!-- List any succinct expected products from this implementation plan. -->
+
+- `openverse.engineering` should move to the Cloudflare free tier
+- All services should live on `openverse.org`
+- Production API domains on `openverse.engineering` should redirect to
+  `openverse.org`
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+### 0. Preliminary work
+
+1. Move the `access` module that configures Cloudflare Access to a new
+   `cloudflare` root module, maintaining the existing configuration.
+   [The rationale for the new root module is explained in this section](#moving-cloudflare-access-and-the-new-cloudflare-root-module).
+1. Extract load balancer listener rules out of `generic/service` and
+   `generic/ec2-service` in favour of a more generic form of the existing
+   `service-alias` called `service-domain` (effects all existing services in the
+   `next` root modules that have external domains). Update all existing services
+   to use this new `service-domain` module for all domains associated with them.
+1. Similarly, extract security group configuration out of `generic/service` and
+   `generic/ec2-service` into a new module, `generic/security-group`. The new
+   module will configure a default security group with strict access based on
+   our load balancer and SSH-bastion entrypoints for HTTP(S) and SSH
+   respectively. As with the previous point, this deduplicates near identical
+   configuration in `generic/service` and `generic/ec2-service`.
+1. Remove `register_with_target_group` option from `ec2-service` to simplify the
+   module. Registration must be handled using Ansible playbooks for these
+   services, so the updated module should never register the instance to the
+   target group.
+
+### 1. Move Airflow to `airflow.openverse.org`
+
+1. Add `airflow.openverse.org` to Cloudflare access.
+1. Create a new `airflow` module in the `next` root module. Use `ec2-service` as
+   the basis with Ansible used to configure the service (compose file, settings,
+   and cron job), to start the service, and to register it with the target
+   group.
+1. Deploy the `next` Airflow module and use the new `service-domain` to point
```

Comment:
```suggestion
1. Spin down the existing Airflow EC2 service to prevent any potential issues
   with multiple running instances.
1. Deploy the `next` Airflow module and use the new `service-domain` to point
```